src/cljam/pileup/pileup.clj

```diff
@@ -2,90 +2,48 @@
   "Provides simple pileup functions."
   (:require [com.climate.claypoole :as cp]
             [cljam.common :refer [get-exec-n-threads]]
+            [cljam.bam :as bam]
             [cljam.util.sam-util :as sam-util]
             [cljam.io :as io]
-            [cljam.pileup.common :refer [window-width step center]]
+            [cljam.pileup.common :as common]
             cljam.bam.reader)
   (:import cljam.bam.reader.BAMReader))
 
-(defn- update-pile
-  "Updates the pile vector from the alignment, returning the updated pile
-  vector. pile must be a vector."
-  [aln rname pile positions]
-  (if (= rname (:rname aln))
-    (let [win-beg (first positions)
-          win-end (last positions)
-          left (max (:pos aln) win-beg)
-          right (min (sam-util/get-end aln) win-end)]
-      (loop [i left, pile* pile]
-        (if (<= i right)
-          (recur (inc i) (update-in pile* [(- i win-beg)] inc))
-          pile*)))
-    (vec (repeat (count pile) 0))))
-
 (defn- count-for-positions
   "Piles the alignments up and counts them in the positions, returning it as a
   vector."
-  [alns rname positions]
-  (loop [[aln & rest] alns
-         pile (vec (repeat (count positions) 0))]
-    (if aln
-      (recur rest (update-pile aln rname pile positions))
-      pile)))
-
-(defn rpositions
-  "Returns a lazy seq of nums from start (inclusive) to end (inclusive)."
-  ([start end]
-     (rpositions start end start))
-  ([start end n]
-     (if (>= end n)
-       (cons n (lazy-seq (rpositions start end (inc n)))))))
-
-(defn grouped-rpositions
-  "Returns a lazy seq of grouped positions. Equivalent of
-  (partition-all n (rpositions start end)). This implementation is faster."
-  [start end n]
-  (lazy-seq
-   (letfn [(rpositions* [s e]
-             (loop [i e xs `()]
-               (if (>= i s)
-                 (recur (dec i) (cons i xs))
-                 xs)))]
-     (when (<= start end)
-       (cons (rpositions* start (min (+ start n -1) end))
-             (grouped-rpositions (+ start n) end n))))))
+  [alns beg end]
+  (let [pile (long-array (inc (- end beg)))]
+    (doseq [aln alns]
+      (let [left (max (:pos aln) beg)
+            right (min (sam-util/get-end aln) end)
+            left-index (- left beg)]
+        (dotimes [i (inc (- right left))]
+          (aset-long pile (+ i left-index) (inc (aget pile (+ i left-index)))))))
+    (seq pile)))
```

Comment:
New `count-for-positions` returns a result as a sequence. Please fix docstring of `count-for-positions`.